src/components/datetime/DateTimeController.cpp

```diff
@@ -117,4 +117,30 @@ char const* DateTime::MonthsString[] = {"--", "JAN", "FEB", "MAR", "APR", "MAY",
 char const* DateTime::MonthsStringLow[] = {"--", "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
 
 char const* DateTime::MonthsLow[] = {
-  "--", "January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"};
\ No newline at end of file
+  "--", "January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"};
+
+void DateTime::GetTimeStr(char* string, bool is24h) {
+  char minutesChar[3];
+  sprintf(minutesChar, "%02d", static_cast<int>(minute));
+  char hoursChar[3];
+  char ampmChar[3] = "  ";
+  if (is24h) {
+    sprintf(hoursChar, "%02d", hour);
+  } else {
+    if (hour == 0) {
+      hour = 12;
```

Comment:
This is a bug. This code was moved from elsewhere in the code where `hour` was a local variable. Here it's the member of `DateTime` so modifying it here actually changes the time other parts of the code see.